src/linear_algebra/affine_space/finite_dimensional.lean

```diff
@@ -254,4 +259,120 @@ lemma findim_vector_span_le_iff_not_affine_independent [fintype ι] (p : ι →
   findim k (vector_span k (set.range p)) ≤ n ↔ ¬ affine_independent k p :=
 (not_iff_comm.1 (affine_independent_iff_not_findim_vector_span_le k p hc).symm).symm
 
+/-- A set of points is collinear if their `vector_span` has dimension
+at most `1`. -/
+def collinear (s : set P) : Prop := dim k (vector_span k s) ≤ 1
+
+/-- The definition of `collinear`. -/
+lemma collinear_iff_dim_le_one (s : set P) : collinear k s ↔ dim k (vector_span k s) ≤ 1 :=
+iff.rfl
+
+/-- A set of points, whose `vector_span` is finite-dimensional, is
+collinear if and only if their `vector_span` has dimension at most
+`1`. -/
+lemma collinear_iff_findim_le_one (s : set P) [finite_dimensional k (vector_span k s)] :
+  collinear k s ↔ findim k (vector_span k s) ≤ 1 :=
+begin
+  have h := collinear_iff_dim_le_one k s,
+  rw ←findim_eq_dim at h,
+  exact_mod_cast h
+end
+
+variables (P)
+
+/-- The empty set is collinear. -/
+lemma collinear_empty : collinear k (∅ : set P) :=
+begin
+  rw [collinear_iff_dim_le_one, vector_span_empty],
+  simp
+end
+
+variables {P}
+
+/-- A single point is collinear. -/
+lemma collinear_singleton (p : P) : collinear k ({p} : set P) :=
```

Comment:
Would it make sense to add another one for pairs `{a,b}`?

Reply:
It's there, further down the file.